### DISCUSSION TITLE
different text colors for single text(dynamic text)

### DISCUSSION THREAD

**Discussion by ashoksrt-248** [clientIO/joint#2737] [Q&A]
### Introduction

I have a Eclipse hook in my application, that hook have dynamic text , i want add a seperate color for when my condition pass(different color for line),actually it is treating entire texts as a single text but i want seperation for each text data

<img width="170" alt="Hook Text color" src="https://github.com/user-attachments/assets/2da02a80-ad39-4b02-b4b8-d518551c89f1">

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You can define a [shape with multiple labels](https://docs.jointjs.com/learn/features/shapes/custom-shapes/#how-can-i-give-multiple-labels-to-an-element) or use [text annotations](https://jsfiddle.net/kumilingus/ngwa29hv/).

**Reply by ashoksrt-248** [reply]
i have dynamic text that is coming from another file

**Reply by ashoksrt-248** [reply]
some times I will have 20+ labels, we are not able to control how many texts it has sometimes it is only having one or two lines, so according to the dynamic texts coming we need to control the each and every line(label text) based upon some condition we want to give some colour

**Reply by kumilingus** [reply]
You can still use **text annotations** and calculate the annotation indexes based on the input text.

Other options are either using ports as shown in this [example](https://www.jointjs.com/demos/list) or the *JointJS+* [Record](https://docs.jointjs.com/learn/features/shapes/built-in-shapes/records) shape.

**Reply by ashoksrt-248** [reply]
that is not working for me

**Reply by kumilingus** [reply]
What exactly isn't working for you?

Please try to elaborate, and provide an example of what you have tried so far.

**Reply by DarkMagus** [reply]
@kumilingus 
I am using Joint JS Plus version 4.0.1 but It does seem that I don't have Record shapes. How to get them? do  I need to use a latest Joint JS plus version?
Thanks

**Reply by DarkMagus** [reply]
@kumilingus 
I found the Record shape and it works well for what I want to accomplish.
Thanks

**Reply by kumilingus** [reply]
It should be available under `shapes.standard.Record`.

```ts
import {  shapes } from '@joint/plus';
const record = new shapes.standard.Record();
```

**Reply by DarkMagus** [reply]
@kumilingus 

Although I am setting the height of a Record node, the height increases with the number of rows.
Is there a way to force a Record to have a fixed height? 
Thanks

**Reply by kumilingus** [reply]
You can set `scrollTop: 0` to disable the automatic resizing (https://docs.jointjs.com/api/shapes/standard/Record/#scrolltop).

**Reply by DarkMagus** [reply]
@kumilingus 
Setting scrollTop does not help as you can see in the below figure where the height is different for three nodes. I set the maximum number of rows to be three, and the third row has ellipsis to show that there are more rows. 
I want the height to be the same for the three nodes whether they have one, two or three rows.


![image](https://github.com/user-attachments/assets/1265442a-4a81-4c62-908b-0b38efeb6591)

**Reply by kumilingus** [reply]
Please see the [DataMapping](https://www.jointjs.com/demos/data-mapping) example on how to use the `scrollTop` attribute.

You can set the record height to any value and the height does not change when adding/removing rows:
<img width="224" alt="image" src="https://github.com/user-attachments/assets/52671d6e-0590-4611-a09e-66d78f5b269e">

**Reply by DarkMagus** [reply]
@kumilingus 
Thanks for the DataMapping exampIe. Using this example, I figured it out that the ```scrollTop``` attribute seems to work only for ```shapes.standard.HeaderedRecord``` but not for ```shapes.standard.Record```

I did this change and the height is fixed now.

Thanks

**Reply by kumilingus** [reply]
It does not matter whether it is a `HeaderedRecord` or just a `Record`. It should work in both cases.

**Reply by DarkMagus** [reply]
@kumilingus 
You are right, I did modify the ```DataMapping``` example from ```HeaderedRecord``` to ```Record``` and it works. 
Something is weird in my code.
I have to read the Record documentation carefully.
Thanks lot.
